.azure-pipelines/ci-build.yml

```diff
@@ -0,0 +1,265 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+name: $(BuildDefinitionName)_$(SourceBranchName)_$(Date:yyyyMMdd)$(Rev:.r)
+
+trigger:
+    branches:
+        include:
+            - main
+pr:
```

Comment:
This configuration will deploy on a PR and a commit to *main*. We don't want to perform a deployment from a PR. Assuming that main is protected, a commit to main after a PR should be the condition for kicking off a build+test+sign+deploy workflow.

Reply:
If the version is bumped in a PR, the pipeline will result in a deployment from the PR. Then, when the PR is merged, it will try to run the pipeline again, but this time, it will stop running once **Validate project version has been incremented** step runs because the version number will have already been bumped. 

The PR should include build, test, and static analysis. The merge into main should trigger everything in the PR and signing and deployment. 

Another note. If the version number is not bumped, we are still using cycles to request strong naming, signing, and NuGet pack before we end the workflow. We should end the workflow earlier.

Reply:
The deploy doesn't happen from a PR. It happens when a branch is merged into main. There's a condition on the deploy stage: https://github.com/microsoftgraph/msgraph-cli-core/blob/07df217d2766b4a99b3246b0189fc3ab14d15d59/.azure-pipelines/ci-build.yml#L236

Reply:
Missed that condition. 

In that case, can you add that condition to all of the tasks after line 108 so that we don't unnecessarily sign, pack, and publish the NuGet artifact, that is, if we aren't validating the signed NuGet artifact for each PR.